op-node/rollup/engine/payload_success.go

```diff
@@ -30,11 +36,17 @@ func (eq *EngDeriver) onPayloadSuccess(ev PayloadSuccessEvent) {
 		})
 	}
 
+	elapsed := time.Since(ev.BuildStarted)
 	payload := ev.Envelope.ExecutionPayload
 	eq.log.Info("Inserted block", "hash", payload.BlockHash, "number", uint64(payload.BlockNumber),
 		"state_root", payload.StateRoot, "timestamp", uint64(payload.Timestamp), "parent", payload.ParentHash,
 		"prev_randao", payload.PrevRandao, "fee_recipient", payload.FeeRecipient,
-		"txs", len(payload.Transactions), "concluding", ev.Concluding, "derived_from", ev.DerivedFrom)
+		"txs", len(payload.Transactions), "concluding", ev.Concluding, "derived_from", ev.DerivedFrom,
+		"build_time", common.PrettyDuration(ev.BuildTime),
+		"import_time", common.PrettyDuration(ev.ImportTime),
+		"total_time", common.PrettyDuration(elapsed),
+		"mgas", float64(payload.GasUsed)/1000000,
+		"mgasps", float64(payload.GasUsed)*1000/float64(elapsed))
 
 	eq.emitter.Emit(TryUpdateEngineEvent{})
```

Comment:
Right now the `total_time` does not include the final `ForkchoiceUpdate` that is part of the block processing procedure. Should we attempt to thread all of the timing field through to the `TryUpdateEngineEvent` and log all the times there instead?

Reply:
How this could be implemented by updating the `TryUpdateEngineEvent` + handler:
```
	case TryUpdateEngineEvent:
		// If we don't need to call FCU, keep going b/c this was a no-op. If we needed to
		// perform a network call, then we should yield even if we did not encounter an error.
		fcuStart := time.Now()
		err := d.ec.TryUpdateEngine(d.ctx)
		fcuTime := time.Since(fcuStart)
		if err != nil && !errors.Is(err, ErrNoFCUNeeded) {
			if errors.Is(err, derive.ErrReset) {
				d.emitter.Emit(rollup.ResetEvent{Err: err})
			} else if errors.Is(err, derive.ErrTemporary) {
				d.emitter.Emit(rollup.EngineTemporaryErrorEvent{Err: err})
			} else {
				d.emitter.Emit(rollup.CriticalErrorEvent{Err: fmt.Errorf("unexpected TryUpdateEngine error type: %w", err)})
			}
		} else {
			elapsed := time.Since(x.BuildStarted)
			payload := x.Envelope.ExecutionPayload
			d.log.Info("Completed processing new L2 unsafe block", "hash", payload.BlockHash, "number", uint64(payload.BlockNumber),
				"state_root", payload.StateRoot, "timestamp", uint64(payload.Timestamp), "parent", payload.ParentHash,
				"prev_randao", payload.PrevRandao, "fee_recipient", payload.FeeRecipient,
				"txs", len(payload.Transactions),
				"build_time", common.PrettyDuration(x.BuildTime),
				"import_time", common.PrettyDuration(x.ImportTime),
				"fcu_time", common.PrettyDuration(fcuTime),
				"total_time", common.PrettyDuration(elapsed),
				"mgas", float64(payload.GasUsed)/1000000,
				"mgasps", float64(payload.GasUsed)*1000/float64(elapsed))
		}
```

Reply:
Trying to interpret the comment `If we don't need to call FCU`. When would we have that scenario?